Server/VikingVault.Services/CompanyService.cs

```diff
@@ -0,0 +1,71 @@
+﻿using Microsoft.EntityFrameworkCore;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using VikingVault.DataAccess;
+using VikingVault.DataAccess.Enums;
+using VikingVault.DataAccess.Models;
+using VikingVault.Services.Abstractions;
+using VikingVault.Services.Exceptions;
+
+namespace VikingVault.Services
+{
+    public class CompanyService : ICompanyService
+    {
+        private readonly VikingVaultDbContext _dbContext;
+        private readonly IBankAccountService _bankAccountService;
+
+        public CompanyService(VikingVaultDbContext dbContext, IBankAccountService bankAccountService)
+        {
+            _dbContext = dbContext;
+            _bankAccountService = bankAccountService;
+        }
+
+        public User CreateCompany(CompanyDTO company)
+        {
+            Role companyRole = _dbContext.Roles.SingleOrDefault(role => role.Type == RoleEnum.Company.ToString());
+            var presentUser = _dbContext.User.SingleOrDefault(user => user.FirstName.ToLower() == company.Name.ToLower());
+            if (presentUser != null)
+            {
+                throw new CompanyServiceException("Company already exists");
+            }
+            User companyUser = new User
+            {
+                FirstName = company.Name,
+                LastName = company.Name,
+                Email = company.Email,
+                Password = company.Name,
+                Address = company.Address,
+                Cnp = "1234567891234",
+                Role = companyRole
+            };
+
+            try
+            {
+                _dbContext.Add(companyUser);
+                _bankAccountService.CreateBankAccount(this.CreateBankAccount(companyUser, "Ron"));
+
+                _dbContext.SaveChanges();
+            } catch (Exception e)
+            {
+                if (e is DbUpdateException || e is DbUpdateConcurrencyException || e is BankAccountServiceException)
```

Comment:
if it's not any of this exceptions what happens? the exception is swallowed and it is hard to figure what went wrong. you should throw regarding the exception type